backend/app.py

```diff
@@ -1,34 +1,135 @@
+                lastname VARCHAR NOT NULL,
+                email VARCHAR NOT NULL,
+                created_at TIMESTAMP WITH TIME ZONE DEFAULT now(),
+                password VARCHAR NOT NULL
+                );
+                """
+            )
+            conn.commit()
+
+    return "created"
+
+
+@app.route("/register", methods=["POST"])
+def register_user():
+    """Register a new user.
+
+    Validates that the username is not already taken.
+    Hashes the password for security.
+    """
+    content_type = request.headers.get("Content-Type")
+    if content_type != "application/json":
+        return "Content-Type not supported!"
+
+    json = request.json
+
+    try:
+        username = json["username"]
+        password = json["password"]
+        firstname = json["firstname"]
+        lastname = json["lastname"]
+        email = json["email"]
+    except KeyError as e:
+        return f"{e} is required."
+
+    error = None
+
+    if not username:
+        error = "Username is required."
+    elif not password:
+        error = "Password is required."
+    elif not firstname:
+        error = "Firstname is required."
+    elif not lastname:
+        error = "Lastname is required."
+    elif not email:
+        error = "Email is required."
+
+    with get_db_connection() as conn:
+        if error is None:
+            try:
+                with conn.cursor() as cur:
+                    cur.execute(
+                        "INSERT INTO users (username, password, firstname, lastname, email) VALUES (%s,%s,%s,%s,%s);",
+                        (
+                            username,
+                            generate_password_hash(password),
+                            firstname,
+                            lastname,
+                            email,
+                        ),
+                    )
+                    conn.commit()
+            except conn.IntegrityError:
+                error = f"User {username} is already registered."
             else:
-                return "<h1>error: No test data found</h1>"
-    finally:
-        conn.close()
+                return f"User {username} was succefull added"
+
+    return f"error: {error}"
+
+
+@app.route("/drop", methods=["POST"])
+def drop_table():
+    content_type = request.headers.get("Content-Type")
+    if content_type != "application/json":
+        return "Content-Type not supported!"
+    json = request.json
+    print(json["table"])
+    table = json["table"]
+
+    error = None
+
+    if not table:
+        error = "table is required."
+
+    conn = get_db_connection()
+
+    with get_db_connection() as conn:
+        if error is None:
+            try:
+                cur = conn.cursor()
+                cur.execute(f"DROP table IF EXISTS {table}")
+                conn.commit()
+                return f'Table "{table}" was succefull dropped'
+            except:
+                error = f"drop database exception"
```

Comment:
```suggestion
                error = "drop database exception"
```
simple string is enough